From 40f7d00c151b343b9d87b6a44563706f3f61e030 Mon Sep 17 00:00:00 2001
From: Cole Robinson <crobinso@redhat.com>
Date: Fri, 6 May 2016 14:03:13 -0400
Subject: [PATCH] ui: gtk: Fix a runtime warning on vte >= 0.37

inner-border was dropped in vte API 2.91, in favor of the standard
padding style

Signed-off-by: Cole Robinson <crobinso@redhat.com>
Message-id: 60a6cdc337d611d902f53907e66a8f37ea374d65.1462557436.git.crobinso@redhat.com

[ kraxel: Fix warning with old vte version. ]

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
(cherry picked from commit 84e2dc4bf39f6ba90390b5f32aadcfcf18168e1d)
[BR: BSC#1008519]
Signed-off-by: Bruce Rogers <brogers@suse.com>
---
 ui/gtk.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/ui/gtk.c b/ui/gtk.c
index 9876d899aa..28e7d28fa8 100644
--- a/ui/gtk.c
+++ b/ui/gtk.c
@@ -330,6 +330,17 @@ static void gd_update_geometry_hints(VirtualConsole *vc)
         VteTerminal *term = VTE_TERMINAL(vc->vte.terminal);
         GtkBorder *ib;
 
+#if VTE_CHECK_VERSION(0, 37, 0)
+        GtkBorder padding;
+        gtk_style_context_get_padding(
+                gtk_widget_get_style_context(vc->vte.terminal),
+                gtk_widget_get_state_flags(vc->vte.terminal),
+                &padding);
+        ib = &padding;
+#else
+        gtk_widget_style_get(vc->vte.terminal, "inner-border", &ib, NULL);
+#endif
+
         geo.width_inc  = vte_terminal_get_char_width(term);
         geo.height_inc = vte_terminal_get_char_height(term);
         mask |= GDK_HINT_RESIZE_INC;
@@ -339,7 +350,7 @@ static void gd_update_geometry_hints(VirtualConsole *vc)
         geo.min_width  = geo.width_inc * VC_TERM_X_MIN;
         geo.min_height = geo.height_inc * VC_TERM_Y_MIN;
         mask |= GDK_HINT_MIN_SIZE;
-        gtk_widget_style_get(vc->vte.terminal, "inner-border", &ib, NULL);
+
         if (ib) {
             geo.base_width  += ib->left + ib->right;
             geo.base_height += ib->top + ib->bottom;
